lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module

```diff
@@ -120,6 +120,15 @@ function dosomething_helpers_extract_field_data($field, $language = LANGUAGE_NON
 
     $data = $field[$language];
 
+    if (!isset($data)) {
+      if (isset($field[LANGUAGE_NONE])) {
+        $data = $field[LANGUAGE_NONE];
+      }
+      else {
+        return;
```

Comment:
definitely `return null;`... so at least you have something to test against in the function that called this.